### PR TITLE
Fix audio snackbar and box audio seeking

### LINKED ISSUE #2124: Boxed audio layout should not be seekable
Currently the boxed audio component, visible on the all results view, is seekable by keyboard (but not by mouse). This is not the intended behavior for the component. It was deemed too small for this seeking functionality to be useful.

We need to make this boxed audio layout not seekable, and make sure it uses the correct `ariaLabel`	in `VAudioTrack`.

### LINKED ISSUE #2742: Audio instructions snackbar is not hidden when you navigate from All results to single result page
## Reproduction

1. Go to https://openverse.org/search/?q=cat
2. Navigate to the first audio result by pressing <kbd>Tab</kbd>. The snackbar should appear.
3. Press Enter to navigate to the single result page.
4. See error: the snackbar is still visible in the corner

## Screenshots

Screenshot 2023-07-29 at 11 55 20 AM

## Environment

Production site, desktop with a keyboard.

## Additional details
I noticed this bug when the Audio single result page visual regression tests failed on #2735

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #2742 by @obulat
Fixes #2124 by @zackkrida 

## Description

To fix the linked issues, this PR refactors the Audio search result components. 

I realized that the components that display the audio result on the All content (`VAudioCell`) and Audio search result pages (the inner part of `VAudioList`) have the same functionality. Both are `<li>` list items, and both have to handle interaction and mouse down to send analytics events. So, I refactored `VAudioCell` to `VAudioResult` that handles that, and can also set the correct audio track layout and size (`row` if it's the Audio search page, and `box` if its the All content search page). 
The new `VAudioResult` component is now used by `VAudioList` (the audio search results page and the related audio section on the single results page) and the `VAllResultsGrid` (the All content search results page).

The **snackbar handling** is moved inside the `VAudioTrack` component. This allows us to not pass as many events up anymore.

We also add `isSeekable` prop to `use-seekable` composable to prevent seeking on box audio by arrow keys.

Another change added here is to hide the related images using `brightness: 0%` and remove the `maxDiffPixels` threshold in the snapshot comparisons. Previously, there was flakiness in these snapshots because sometimes the related images did not fully load and still had gray spans when we took the snapshots. This is not related to this PR, but since this PR was started because the Playwright failures in #2735, I decided to add it here, and rebase 2735 on this PR. This will help us see the changes in the external links in that PR.

## Testing Instructions

Test that the instructions snackbar is closed when you navigate away from search results. Go to the search page, navigate to the Audio item using keyboard, and close the snackbar by pressing Spacebar to play the audio. Then, press Enter to navigate to the single result page. The snackbar should be hidden.

Test that you cannot seek on the box layout. Go to the all content search page, and navigate to an audio item using your keyboard. Press arrow keys: you shouldn't be able to "seek" the audio. You should also see no `AUDIO_INTERACTION` analytics events in the console.

Check that audio interaction and `SELECT_SEARCH_RESULT` events are sent correctly for all audio tracks. When you click on audio to navigate to the single result page, you should get the `SELECT_SEARCH_RESULT` but not `AUDIO_INTERACTION` event. When you click play or seek - you should get the `AUDIO_INTERACTION`, but no `SELECT_SEARCH_RESULT` events.

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by sarayourfriend — APPROVED** [review]
Works when I test it locally :rocket: Just one clarification question to understand the intended interaction.

File: frontend/src/components/VAudioTrack/VAudioTrack.vue [hunk 1]
```diff
@@ -142,6 +143,15 @@ export default defineComponent({
...
     },
+    /**
+     * Whether instructions snackbar should be shown when this track is focused.
+     * This is used to show instructions for keyboard navigation on the search
+     * results page.
+     */
+    showsSnackbar: {
+      type: Boolean,
+      default: false,
+    },
```
[the remark this answers is not in the document]
Comment (obulat) [reply frontend/src/components/VAudioTrack/VAudioTrack.vue]: I've asked @fcoveram about it, and we decided that we want to show the instructions for _all_ audio tracks (even the main audio track for the single result page that didn't use to show it), and to hide it when the audio track loses focus. 
I updated the PR to reflect that. So, now when you focus on any audio track, you see the instructions snackbar. It is completely dismissed if you interact with the audio using your keyboard. If you move the focus away from the audio track, the snackbar is only hidden (so, it re-appears if you focus on an audio track again).

[the remark this answers is not in the document]
Comment (zackkrida) [reply frontend/src/components/VAudioTrack/VAudioTrack.vue]: How does mobile factor in here? I just noticed on the live site that clicking the boxed audio layout shows the snack bar on mobile

**Review by zackkrida — APPROVED** [review]
LGTM. The [question I asked about mobile](https://github.com/WordPress/openverse/pull/2743#discussion_r1289389631) can perhaps be handled, if necessary, in a separate issue.

**Pull request merged**